src/controllers/task.js

```diff
@@ -202,6 +202,61 @@ const TaskController = (taskModel, userModel, authService, googleAPIService) =>
         })
     });
 
+    router.patch('/:id', async (req, res) => {
+        const params = req.params;
+        const id = parseInt(params.id, 10);
+
+        // Get the user_id of the user sending the request
+        const [user_id_from_request, err1] = await authService.getLoggedInUserID(req.headers);
+        if (err1) {
+            return res.status(400).json({
+                message: err1.message,
+            });
+        }
+        const [task, err2] = await taskModel.getTask(id);
+        if (!req.body)
+            return res.status(400).json({
+                data: null,
+                error: "Malformed Request"
+            });
+        if (user_id_from_request != task.user_id) {
+            // Make sure the requestor has access to this object, if not, Access Denied
+            return res.status(403).json({
+                data: {},
+                error: "Access Denied"
+            });
+        }
+        // get all the info from what the user wants to edit and register it into dictionary
+        const updatedTask = {};
+        const body = req.body;
+        if (body.name)
+            updatedTask["name"] = body.name;
+        if (body.description)
+            updatedTask["description"] = body.description;
+        if (body.duration)
+            updatedTask["duration"] = body.duration;
+        if (body.due_date)
+            updatedTask["due_date"] = body.due_date;
+        // call function to edit with the dictionary
+        var str = [];
+        for (var key in updatedTask)
+            if (updatedTask.hasOwnProperty(key)) {
+                str.push(key + "=" + "\'" + updatedTask[key] + "\'");
```

Comment:
Will this add quotations around the duration value and make it a string instead of an int?

Reply:
If this doesnt cause issues then you don't need to fix it, I'm just flagging to make sure you check